Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,30 +188,111 @@ void HiresTexture::Update()
+               percent_mem_to_cull);
+    }
+  }
+
+  double prefetched_size_sum = 0;  // total mem size incl. non-native decompressed textures
+  const double absolute_max_mem = Common::MemPhysical();
+  // Currently "total usable", excl. HW reserved
+  const double culled_percent_amount = (Common::MemPhysical() * percent_mem_to_cull) / 100;
+  const double delegated_max_mem = Common::MemPhysical() - culled_percent_amount;
+  const double below_min_req_mem = 3221225472;  // 3 GB; 2GB Rec. + 1GB OS/Misc
+  // WIP - using 1024 * ... etc but makes the line so long, no global ref found, make one?
+
+  // message only
+  if (absolute_max_mem <= below_min_req_mem)
+  {
+    OSD::AddMessage("Prefetcher: Warning! System RAM of " +
+                        StringFromFormat("%.2f MB", absolute_max_mem / (1024.0 * 1024.0)) +
+                        " is below Dolphin's minimum recommended specifications.",
+                    20000);
+    WARN_LOG(VIDEO, "Usable system memory below Dolphin's minimum recommended specifications");
+  }
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  // Routine Stat Notification
+  OSD::AddMessage("Prefetcher: Maximum delegated memory limit has been configured to " +
+                      StringFromFormat("%.2f MB", delegated_max_mem / (1024.0 * 1024.0)) +
+                      ", using cull of " + StringFromFormat("%.1f %%", percent_mem_to_cull),
+                  20000);
+
+  // Start Prefetching
+  prefetch_starttime_total = Common::Timer::GetTimeMs();
+
+  for (const auto& texmap_ref : s_textureMap)
   {
-    const std::string& base_filename = entry.first;
+    const std::string& base_filename = texmap_ref.first;
+    // const std::string& full_filepath = texmap_ref.second;
+    // idea to calc m_levels data_size per extension, but existing system uses base filenames
+    // would add ability to calc non-native texture(PNG) in mem and compare to ondisk size
+    // with the right formula that would return the amount of overhead
+    // (for 14 Megs of PNGs mem is ~350MB)
+    // to get the per-extension time probably this whole for loop would have to be redone
+    // I was not able to do this with full day hours while leaving "m_levels" unchanged
 
     if (base_filename.find("_mip") == std::string::npos)
     {
       {
         // try to get this mutex first, so the video thread is allow to get the real mutex faster
+        // JonnyH: so the prefetch() is running in a tight loop, and it's using that to allow
+        // "at least" 1 Search() run per cycle
+        // merry: exactly!
```

Comment:
> I asked the IRC what to do about this, they provided none, which to me from now on should be equal to "not bad enough for them to care stopping the pressed and replying"

It's not equal to that.

I think many people on IRC are choosing to not respond to you because discussions with you tend to be a waste of time. That doesn't mean that we don't care about not making the source code worse.

> so I had to come up with a solution my self.

A solution for what? It wasn't broken, and I can't see anything you changed other than adding this comment.

> or I'm have to going to sit in IRC for hours to get things pre-approved

I don't recommend sitting in IRC for hours to get things approved. What I recommend is not making any PRs if they quality of them is going to be as bad as this one. We've already spent countless of hours on helping you on IRC, and most of us probably aren't willing to put much more time into helping you with every detail of every PR you're making.